core/variant_op.cpp

```diff
@@ -678,6 +678,7 @@ void Variant::evaluate(const Operator &p_op, const Variant &p_a,
 			DEFAULT_OP_LOCALMEM(math, OP_LESS, VECTOR2I, <, Vector2i);
 			DEFAULT_OP_LOCALMEM(math, OP_LESS, VECTOR3, <, Vector3);
 			DEFAULT_OP_LOCALMEM(math, OP_LESS, VECTOR3I, <, Vector3i);
+			DEFAULT_OP_LOCALMEM(math, OP_LESS, STRING_NAME, <, StringName);
```

Comment:
I see that there's `DEFAULT_OP_STR*` macros which are supposed to handle `StringName` too - maybe they should be used here?

I also wonder whether `NodePath` has the same issue.

Reply:
NodePath works fine.

Reply:
SringNames are not necesarily strings, so technically only ==, != and should be implemented imo,and < for sorting.